### ISSUE TITLE
gui: crash in Console after loading & unloading wallet

### ISSUE THREAD

**Issue by fanquake** [bitcoin/bitcoin#16306]
Using the Console, this crash occurs after loading a wallet (which starts rescanning), then unloading the same wallet. Using master (0f309541aa6dee91ac1cf6058c1ab5f51447b3c1).

```bash
./autogen.sh && ./configure --enable-debug && make check -j8 && make deploy
```

```bash
lldb Bitcoin-Qt.app
(lldb) target create "Bitcoin-Qt.app"
Current executable set to 'Bitcoin-Qt.app' (x86_64).
(lldb) run
Process 8595 launched: '/Users/michael/github/bitcoin/Bitcoin-Qt.app/Contents/MacOS/Bitcoin-Qt' (x86_64)
2019-06-29 12:51:35.076094+0800 Bitcoin-Qt[8595:1670460] MessageTracer: Falling back to default whitelist
Process 8595 stopped
* thread #1, name = 'bitcoin-main', queue = 'com.apple.main-thread', stop reason = EXC_BAD_ACCESS (code=EXC_I386_GPFLT)
    frame #0: 0x0000000101f1f89d QtCore`QObject::connect(QObject const*, char const*, QObject const*, char const*, Qt::ConnectionType) + 125
QtCore`QObject::connect:
->  0x101f1f89d <+125>: movq   (%r14), %rax
    0x101f1f8a0 <+128>: movq   %r14, %rdi
    0x101f1f8a3 <+131>: callq  *(%rax)
    0x101f1f8a5 <+133>: movq   %r12, -0x1a8(%rbp)
Target 0: (Bitcoin-Qt) stopped.
```

```bash
2019-06-29T04:52:35Z UpdateTip: new best=00000000000001fae6c844a0973ca2b25abbfa5ceabd7ad1f29a30169f24771e height=176516 version=0x00000001 log2_work=68.020722 tx=2878576 date='2012-04-20T23:28:53Z' progress=0.006761 cache=46.4MiB(353740txo)
2019-06-29T04:52:35Z Using BerkeleyDB version Berkeley DB 4.8.30: (April  9, 2010)
2019-06-29T04:52:35Z Using wallet /Users/michael/Library/Application Support/Bitcoin/some wallet
2019-06-29T04:52:35Z BerkeleyEnvironment::Open: LogDir=/Users/michael/Library/Application Support/Bitcoin/some wallet/database ErrorFile=/Users/michael/Library/Application Support/Bitcoin/some wallet/db.log
2019-06-29T04:52:35Z UpdateTip: new best=00000000000002229dcf910627a2c8e64885390e7e3bded45bb38149f36ec108 height=176517 version=0x00000001 log2_work=68.020754 tx=2878644 date='2012-04-20T23:42:26Z' progress=0.006761 cache=46.4MiB(353704txo)
2019-06-29T04:52:36Z UpdateTip: new best=0000000000000242d1e72b1b06aa731ed516ab1119223570850da9cc8b36f4ea height=176518 version=0x00000001 log2_work=68.020787 tx=2878704 date='2012-04-20T23:43:26Z' progress=0.006761 cache=46.4MiB(353824txo)
2019-06-29T04:52:36Z UpdateTip: new best=0000000000000202e681cb9e07b4a5c63869153a865d30d02f283d098cdbf186 height=176519 version=0x00000001 log2_work=68.02082 tx=2878740 date='2012-04-20T23:48:38Z' progress=0.006761 cache=46.4MiB(353790txo)
2019-06-29T04:52:36Z UpdateTip: new best=0000000000000391bc5a7df2aca4ee4eed786ea16a3f76e1ae7030cd82a6fd1b height=176520 version=0x00000001 log2_work=68.020852 tx=2878866 date='2012-04-20T23:56:11Z' progress=0.006761 cache=46.4MiB(353854txo)
2019-06-29T04:52:36Z UpdateTip: new best=00000000000000d664f0b5ddd5617dcd8ec5010f7d5745aa2563a9beefd4036f height=176521 version=0x00000001 log2_work=68.020885 tx=2878940 date='2012-04-21T00:06:06Z' progress=0.006762 cache=46.4MiB(353939txo)
2019-06-29T04:52:36Z UpdateTip: new best=00000000000009c4df44f6e721b7e775953da05713d70d4d470776453535abab height=176522 version=0x00000001 log2_work=68.020918 tx=2878970 date='2012-04-21T00:05:26Z' progress=0.006762 cache=46.4MiB(354025txo)
2019-06-29T04:52:36Z UpdateTip: new best=000000000000002f01198e0fd214566715ad4c78d6ee6c7f325d35670a4c250a height=176523 version=0x00000001 log2_work=68.02095 tx=2878971 date='2012-04-21T00:09:46Z' progress=0.006762 cache=46.4MiB(354026txo)
2019-06-29T04:52:36Z UpdateTip: new best=00000000000008b6cee08dc8ddbf3930d12c3d2796309fa21f22369c666a1e3d height=176524 version=0x00000001 log2_work=68.020983 tx=2879001 date='2012-04-21T00:13:03Z' progress=0.006762 cache=46.4MiB(354026txo)
2019-06-29T04:52:36Z UpdateTip: new best=00000000000007f68e267217145d0582a12ae5bc17c35b9f1f768fc8161fe972 height=176525 version=0x00000001 log2_work=68.021016 tx=2879017 date='2012-04-21T00:15:51Z' progress=0.006762 cache=46.4MiB(354035txo)
2019-06-29T04:52:36Z init message: Loading wallet...
2019-06-29T04:52:36Z BerkeleyEnvironment::Open: LogDir=/Users/michael/Library/Application Support/Bitcoin/some wallet/database ErrorFile=/Users/michael/Library/Application Support/Bitcoin/some wallet/db.log
2019-06-29T04:52:36Z UpdateTip: new best=00000000000007fb9d3d05eb42930e344c5bea9cc2eabbc12013607f316b2e2f height=176526 version=0x00000001 log2_work=68.021048 tx=2879089 date='2012-04-21T00:34:08Z' progress=0.006762 cache=46.4MiB(353993txo)
2019-06-29T04:52:36Z UpdateTip: new best=000000000000055bb0ac3d1d33c48f8288f9725bcbed6555eebcafd5031f217a height=176527 version=0x00000001 log2_work=68.021081 tx=2879132 date='2012-04-21T00:38:43Z' progress=0.006762 cache=46.4MiB(353997txo)
2019-06-29T04:52:36Z [some wallet] nFileVersion = 189900
2019-06-29T04:52:36Z [some wallet] Keys: 0 plaintext, 2001 encrypted, 2001 w/ metadata, 2001 total. Unknown wallet records: 0
2019-06-29T04:52:36Z [some wallet] Wallet completed loading in             128ms
2019-06-29T04:52:36Z init message: Rescanning...
2019-06-29T04:52:36Z [some wallet] Rescanning last 5406 blocks (from block 171121)...
2019-06-29T04:52:36Z [some wallet] Rescan started from block 000000000000030de70be2d0a247608bd5ce681744076ccda1a3d24dab18bb65...
2019-06-29T04:52:40Z [some wallet] Rescan completed in            4175ms
2019-06-29T04:52:40Z [some wallet] setKeyPool.size() = 2000
2019-06-29T04:52:40Z [some wallet] mapWallet.size() = 0
2019-06-29T04:52:40Z [some wallet] mapAddressBook.size() = 0
2019-06-29T04:52:40Z UpdateTip: new best=000000000000022f132ffe582f0156576fb4834ecf29abc08ff387ef2e56b31b height=176528 version=0x00000001 log2_work=68.021114 tx=2879222 date='2012-04-21T00:56:16Z' progress=0.006762 cache=46.4MiB(354000txo)
2019-06-29T04:52:40Z [some wallet] Releasing wallet
```

**Comment by promag** [comment]
@fanquake this might be fixed by #18338.

**Comment by MarcoFalke** [comment]
Is this fixed?
